mgmtsystem/azure.py

```diff
@@ -96,21 +98,42 @@ def stop_vm(self, vm_name, resource_group=None, shutdown=False):
             self.run_script(
                 "Stop-AzureRmVm -ResourceGroup \"{}\" -Name \"{}\" -Force"
                 .format(resource_group or self.resource_group, vm_name).strip())
+            self.wait_vm_stopped(vm_name, resource_group)
         else:
             self.logger.info("Azure VM {} is already stopped".format(vm_name))
 
+    def suspend_vm(self, vm_name, resource_group=None):
+        if self.is_vm_running(vm_name, resource_group or self.resource_group):
+            self.logger.info("Attempting to Suspend Azure VM {}".format(vm_name))
+            self.run_script(
+                "Stop-AzureRmVm -ResourceGroup \"{}\" -Name \"{}\" -Force -StayProvisioned"
+                .format(resource_group or self.resource_group, vm_name).strip())
+            self.wait_vm_suspended(vm_name, resource_group)
+        else:
+            self.logger.info("Azure VM {} is already suspended or stopped".format(vm_name))
+
     def wait_vm_running(self, vm_name, resource_group=None, num_sec=300):
         wait_for(
             lambda: self.is_vm_running(vm_name, resource_group or self.resource_group),
             message="Waiting for Azure VM {} to be running.".format(vm_name),
             num_sec=num_sec)
 
+    def wait_vm_steady(self, vm_name, resource_group=None, num_sec=300):
+        self.logger.info("All states are steady in Azure. {}".format(vm_name))
```

Comment:
This wait serves to make sure that a VM / instance is `running`, `stopped`, `suspended` and we can work with it meaning it should exclude stuff like `starting`, `suspending`, `stopping`, `terminating` etc and it leverages the below-defined `in_steady_state` for that. You are overriding [this](https://github.com/ManageIQ/wrapanapi/blob/1f61b0f72adbefb5cd434385ccfc084321c4f37d/mgmtsystem/base.py#L300) method from the `Base` class and I'm not sure I understand why.